### PR TITLE
Upgrade to latest JUnit-Foundation

### LINKED ISSUE #59: Execution crashes when test class is marked as @Ignore
RP is able to successfully skip tests(methods) annotated with @Ignore however classes with same annotations causes NullPointerException:

> [ERROR] org.apache.maven.surefire.testset.TestSetFailedException: java.lang.NullPointerException
[ERROR] 	at org.apache.maven.surefire.common.junit4.JUnit4RunListener.rethrowAnyTestMechanismFailures(JUnit4RunListener.java:209)
[ERROR] 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:169)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:290)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:242)
[ERROR] 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:121)
[ERROR] Caused by: java.lang.NullPointerException
[ERROR] 	at com.epam.reportportal.junit.ParallelRunningHandler.buildStartTestItemRq(ParallelRunningHandler.java:307)
[ERROR] 	at com.epam.reportportal.junit.ParallelRunningHandler.handleTestSkip(ParallelRunningHandler.java:190)
[ERROR] 	at com.epam.reportportal.junit.ReportPortalListener.testIgnored(ReportPortalListener.java:97)
[ERROR] 	at com.nordstrom.automation.junit.RunAnnouncer.testIgnored(RunAnnouncer.java:92)
[ERROR] 	at org.junit.runner.notification.SynchronizedRunListener.testIgnored(SynchronizedRunListener.java:77)
[ERROR] 	at org.junit.runner.notification.RunNotifier$6.notifyListener(RunNotifier.java:174)
[ERROR] 	at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:72)
[ERROR] 	at org.junit.runner.notification.RunNotifier.fireTestIgnored(RunNotifier.java:171)
[ERROR] 	at org.junit.internal.builders.IgnoredClassRunner.run(IgnoredClassRunner.java:16)
[ERROR] 	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:367)
[ERROR] 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:274)
[ERROR] 	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
[ERROR] 	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:161)
[ERROR] 	... 3 more

### PR DESCRIPTION
**Pull request by sbabcoc**
Fixes #59

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -42,11 +42,7 @@ dependencies {
         compile 'com.github.reportportal:commons-model:e608d19'
     }
 
-    compile('com.nordstrom.tools:junit-foundation:11.3.0') {
-        exclude group: 'commons-beanutils', module: 'commons-beanutils'
-    }
-    compile 'commons-beanutils:commons-beanutils:1.9.4'
-
+    compile 'com.nordstrom.tools:junit-foundation:12.0.0'
```
Comment (sbabcoc) [inline build.gradle:45]: Note that this release of **JUnit Foundation** includes a breaking change in the `ArtifactParams` feature. The version assigned to `agent-java-junit` should reflect this change (i.e. - increment the major version).

**Comment by sbabcoc** [comment]
It seems that there are fundamental issues with the v5 agent. Do we have a design document that specifies the intended association between JUnit Foundation notifications and ReportPortal messages? I'd like to help get this v5 agent across the finish line, but I need to know what I'm shooting for.

**Comment by DzmitryHumianiuk** [comment]
@HardNorth will have detailed look in a while

File: build.gradle (hunk 1 above)
Comment (HardNorth) [reply build.gradle]: Well, I believe in the future 5.0.0 release we can introduce these changes.

**Pull request merged**